client/src/components/SocialMediaForm.js

```diff
@@ -20,31 +21,54 @@ const SocialMediaForm = () => {
 
   const handleSubmit = async (event) => {
     event.preventDefault();
-    const response = await fetch("/api/v1/websites", {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify(formData),
-    });
-    if (response.ok) {
-      setFormData({
-        name: "",
-        url: "",
-        description: "",
-        imageUrl: "",
+    let submitErrors = {};
+    if (formData.name.trim() === "") {
+      submitErrors = {
+        ...submitErrors,
+        name: "cannot be empty",
+      };
+    }
+    if (formData.url.trim() === "") {
+      submitErrors = {
+        ...submitErrors,
+        url: "cannot be empty",
+      };
+    }
+    if (formData.description.trim() === "") {
+      submitErrors = {
+        ...submitErrors,
+        description: "cannot be empty",
+      };
+    }
+    if (formData.imageUrl.trim() === "") {
+      submitErrors = {
+        ...submitErrors,
+        imageUrl: "cannot be empty",
+      };
+    }
+
+    setErrors(submitErrors);
+
+    if (Object.keys(submitErrors).length === 0) {
+      const response = await fetch("/api/v1/websites", {
```

Comment:
fetch requests need to be wrapping in a try/catch block and include error handling for any errors the server sends back. You do have front-end validations which is great, but we need to validate the data each step of the way and account for potential errors from the back-end. This is particularly important when we are dealing with information that needs to be unique. The front-end here will not be able to validate if the submitted objects passes for uniqueness, only the back-end can check that and we need to be able to display any potential server errors. This means you will also want to make sure the router for this POST endpoint will appropriately handle ValidationErrors